docs/design-docs/infrastructure/naming-convention.md

```diff
@@ -0,0 +1,40 @@
+# Cloud resources naming convention
+
+This document describes recommendations how to name infrastructure resources that are usually created by Terraform.
+Unifying resource names allows easily identify and search for any resource even if no specific tags were provided.
+
+Listed points are based on development of Epiphany modules
+and [best practices](https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-naming)
+provided by Microsoft Azure.
+
+In general resource name should match following schema:
+
+`<prefix>-<resource_type>-<index>`
+
+or
+
+`<prefix>-<vm_group>-<resource_type>-<index>` when resource name is related to VM group.
+
+## Prefix
+
+Epiphany modules are developed in the way that allows user specify a prefix for created resources. This approach gives
+such benefits as ordered sorting and identifying who is the owner of the resource. Prefix can include following parts
+with a dash `-` as a delimiter.
+
+| Type | Required | Description | Examples |
+| ---- | -------- | ----------- | -------- |
+| Owner | yes | The name of the person or team which resource belongs to | epiphany |
+| Application or service name | yes | Name of the application, workload, or service that the resource is a part of | kafka, ignite, opendistro |
```

Comment:
One more question regarding required=yes here. I'm not sure, but that doesn't look like required to me. Is resource group part of some application for example? I believe owner is one required field for sure. 

At second look at this document it also looks that vm_group is just one of elements of prefix.

Reply:
Please check now